evmrpc/tx.go

```diff
@@ -53,6 +53,44 @@ func (t *TransactionAPI) GetTransactionReceipt(ctx context.Context, hash common.
 		}
 		return nil, err
 	}
+	// Fill in the receipt if the transaction has failed and used 0 gas
+	if receipt.Status == 0 && receipt.GasUsed == 0 {
```

Comment:
can you add a comment here saying that this case is for when a tx fails before it makes it to the VM?